src/test/java/jewellery/inventory/unit/service/SaleServiceTest.java

```diff
@@ -180,20 +176,19 @@ void testReturnProductWillThrowsSaleNotFoundException() {
 
   @Test
   void testReturnProductSuccessfullyWithTwoProducts() {
-    product.setPartOfSale(saleTwoProducts);
+    product.setPartOfSale(sale);
     Product productBeforeReturn = product;
 
-    assertEquals(2, saleTwoProducts.getProducts().size());
+    assertEquals(1, sale.getProducts().size());
     assertNotNull(productBeforeReturn.getPartOfSale());
 
     when(productService.getProduct(any(UUID.class))).thenReturn(product);
-    when(saleRepository.findById(any(UUID.class))).thenReturn(Optional.of(saleTwoProducts));
+    when(saleRepository.findById(any(UUID.class))).thenReturn(Optional.of(sale));
     when(saleService.returnProduct(product.getId())).thenReturn(productReturnResponseDto);
 
     ProductReturnResponseDto productReturnResponseDto = saleService.returnProduct(product.getId());
 
-    assertEquals(1, productReturnResponseDto.getSaleAfter().getProducts().size());
-    assertNull(productReturnResponseDto.getSaleAfter().getId());
```

Comment:
This looks suspicious because we are expecting the sale to not be deleted after returning 1 out of 2 products.